### PR TITLE
Refactor tokens section in checker.yaml

### PR DESCRIPTION
**Pull request by dorranh**
This PR takes a first step towards addressing #291 by refactoring the `tokens` section of `checker.yaml` and corresponding build script logic, adding fields for `wtez`, `wctez`, `mock_fa2`. It also reorganizes the `tokens` schema a bit, distinguishing between `issued` and `in_use` tokens since tokens in the former category need to allow for configurable FA2 metadata.

I also went ahead and added a field for the `collateral_type` switch and made the build script throw an error if it is ever not set to `tez` since we haven't implemented the FA2 build logic yet.

After this PR I should be able to implement the generation of a new `tokenMetadata.ml` module we discussed which can be used within `tezWrapper.ml` and `wctez.ml` (and other generated token modules).

### CODE REVIEW

File: scripts/builder/checker_builder/config.py [hunk 1]
```diff
@@ -114,10 +143,37 @@ class Constants:
...
+        # Validation logic
+        wtez = self.tokens.issued.wtez
+        wctez = self.tokens.issued.wctez
+        collateral = self.tokens.in_use.collateral
+        cfmm_token = self.tokens.in_use.cfmm_token
+
+        if self.collateral_type == CollateralType.TEZ:
+            if not (collateral.token_id == wtez.token_id) and (
+                collateral.decimal_digits == wtez.decimal_digits
+            ):
```
Comment (gkaracha) [inline scripts/builder/checker_builder/config.py:161]: Hmmm, shouldn't the not scope over the whole condition? I mean like this:

```suggestion
            if not ((collateral.token_id == wtez.token_id) and (
                collateral.decimal_digits == wtez.decimal_digits
            )):
```

File: scripts/builder/checker_builder/config.py [hunk 2]
```diff
@@ -114,10 +143,37 @@ class Constants:
...
+                raise ValueError(
+                    "collateral config must be identical to wtez config when collateral_type=tez"
+                )
+            if not (cfmm_token.token_id == wctez.token_id) and (
+                cfmm_token.decimal_digits == wctez.decimal_digits
+            ):
```
Comment (gkaracha) [inline scripts/builder/checker_builder/config.py:167]: Same here :thinking:

**Review by gkaracha — COMMENTED** [review]
Looks good to me overall, except for the negation in the if-then-else conditions (see below); lemme know if I got that wrong :sweat_smile: No idea what's up with CI but let's not merge before that is resolved.

File: scripts/builder/checker_builder/config.py (hunk 1 above)
Comment (dorranh) [reply scripts/builder/checker_builder/config.py:161]: Ah yes it absolutely should. Good catch!

File: checker.yaml [hunk 3]
```diff
@@ -1,50 +1,83 @@
...
+      token_id: 2
+      # Specifies the number of decimal digits used by the token
+      decimal_digits: 6
+    # CFMM token (creates ctok.ml).
+    # Note: Must be configured with same values as wctez when collateral_type=tez
+    cfmm_token:
+      # The token's FA2 id
+      token_id: 42
+      # Specifies the number of decimal digits used by the token
+      decimal_digits: 6
```
[the remark this answers is not in the document]
Comment (dorranh) [reply checker.yaml:80]: I think that for the current configuration with `collateral_type: tez` this has to match `wctez`. Definitely agree for once we have implement support for `fa2` collateral though.

[the remark this answers is not in the document]
Comment (gkaracha) [reply checker.yaml:80]: Ah, sorry, I didn't mean this one here, no idea how I ended up leaving this comment :facepalm: Scrap it.

**Review by gkaracha — APPROVED** [review]
LGTM! :+1:

**Pull request merged**